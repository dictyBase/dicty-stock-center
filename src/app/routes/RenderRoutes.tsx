import React, { lazy, Suspense } from "react"
import { Route, Switch } from "react-router-dom"
import PrivateRoute from "./PrivateRoute"
import Loader from "common/components/Loader"
import useGoogleAnalytics from "common/hooks/useGoogleAnalytics"

const Homepage = lazy(
  () => import(/* webpackChunkName: "Homepage" */ "features/Home/Homepage"),
)

// authentication routes
const Login = lazy(
  () => import(/* webpackChunkName: "Login" */ "features/Authentication/Login"),
)
const OauthCallback = lazy(
  () =>
    import(
      /* webpackChunkName: "OauthCallback" */ "features/Authentication/OauthCallback"
    ),
)
const AuthLoader = lazy(
  () =>
    import(
      /* webpackChunkName: "AuthLoader" */ "features/Authentication/AuthLoader"
    ),
)
const Logout = lazy(
  () =>
    import(/* webpackChunkName: "Logout" */ "features/Authentication/Logout"),
)
const MyDscPage = lazy(
  () => import(/* webpackChunkName: "MyDscPage" */ "features/MyDsc/MyDscPage"),
)

// editable page routes
const InformationContainer = lazy(
  () =>
    import(
      /* webpackChunkName: "Information" */ "features/EditablePages/Information"
    ),
)
const InfoPageContainer = lazy(
  () =>
    import(
      /* webpackChunkName: "InfoPageContainer" */ "features/EditablePages/InfoPageContainer"
    ),
)
const EditInfoPage = lazy(
  () =>
    import(
      /* webpackChunkName: "EditInfoPage" */ "features/EditablePages/EditInfoPage"
    ),
)

const AddPage = lazy(
  () =>
    import(
      /* webpackChunkName: "EditInfoPage" */ "features/EditablePages/AddPage"
    ),
)

// strain routes
const StrainCatalogWrapper = lazy(
  () =>
    import(
      /* webpackChunkName: "StrainCatalogWrapper" */ "features/Stocks/Catalogs/common/CatalogWrapper"
    ),
)
const StrainDetailsContainer = lazy(
  () =>
    import(
      /* webpackChunkName: "StrainDetailsContainer" */ "features/Stocks/Details/Strains/StrainDetailsContainer"
    ),
)

// phenotype routes
const PhenotypesWrapper = lazy(
  () =>
    import(
      /* webpackChunkName: "PhenotypesWrapper" */ "features/Stocks/SearchResults/PhenotypeContainer"
    ),
)

// plasmid routes
const PlasmidCatalogWrapper = lazy(
  () =>
    import(
      /* webpackChunkName: "PlasmidCatalogWrapper" */ "features/Stocks/Catalogs/common/CatalogWrapper"
    ),
)
const PlasmidDetailsContainer = lazy(
  () =>
    import(
      /* webpackChunkName: "PlasmidDetailsContainer" */ "features/Stocks/Details/Plasmids/PlasmidDetailsContainer"
    ),
)

// order form routes
const OrderForm = lazy(
  () =>
    import(/* webpackChunkName: "OrderForm" */ "features/OrderForm/OrderForm"),
)
const OrderConfirmation = lazy(
  () =>
    import(
      /* webpackChunkName: "OrderConfirmation" */ "features/OrderForm/OrderConfirmation"
    ),
)

// misc routes
const ContactPage = lazy(
  () =>
    import(
      /* webpackChunkName: "ContactPage" */ "features/Contact/ContactPage"
    ),
)
const ShoppingCartPage = lazy(
  () =>
    import(
      /* webpackChunkName: "ShoppingCartPage" */ "features/ShoppingCart/ShoppingCartPage"
    ),
)

// custom 404 route
const PageNotReady = lazy(
  () =>
    import(
      /* webpackChunkName: "PageNotReady" */ "features/Errors/PageNotReady"
    ),
)

const RenderRoutes = () => {
  useGoogleAnalytics()

  return (
    <Suspense fallback={<Loader />}>
      <Switch>
        <Route exact path="/" component={Homepage} />
        {/* authentication routes */}
        <Route exact path="/login" component={Login} />
        <Route exact path="/:provider/callback" component={OauthCallback} />
        <Route exact path="/load/auth" component={AuthLoader} />
        <PrivateRoute exact path="/logout" component={Logout} />
        <PrivateRoute exact path="/mydsc" component={MyDscPage} />
        {/* editable page routes */}
        <Route exact path="/information" component={InformationContainer} />
        <Route exact path="/information/:name" component={InfoPageContainer} />
        <PrivateRoute
          exact
          path="/information/:name/edit"
          component={EditInfoPage}
        />
        <PrivateRoute exact path="/addpage" component={AddPage} />
        {/* order form routes */}
        <Route exact path="/order" component={OrderForm} />
        <Route exact path="/order/checkout" component={OrderForm} />
        <Route exact path="/order/submitted" component={OrderConfirmation} />
        {/* strain routes */}
        <Route
          exact
          path="/strains"
          render={(props) => (
            <StrainCatalogWrapper {...props} stockType="strain" />
          )}
        />
        <Route exact path="/strains/:id" component={StrainDetailsContainer} />
        {/* phenotype routes */}
        <Route exact path="/phenotypes/:name" component={PhenotypesWrapper} />
        {/* plasmid routes */}
        <Route
          exact
          path="/plasmids"
          render={(props) => (
            <PlasmidCatalogWrapper {...props} stockType="plasmid" />
          )}
        />
        <Route exact path="/plasmids/:id" component={PlasmidDetailsContainer} />
        {/* misc routes */}
        <Route exact path="/contact" component={ContactPage} />
        <Route exact path="/cart" component={ShoppingCartPage} />
        <Route exact path="*" component={PageNotReady} />
      </Switch>
    </Suspense>
  )
}

export default RenderRoutes
