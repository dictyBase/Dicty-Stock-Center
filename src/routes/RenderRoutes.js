import React from "react"
import { Route, Switch } from "react-router-dom"
import Loadable from "react-loadable"
import PrivateRoute from "./PrivateRoute"
import LoginRoute from "./LoginRoute"
import Loader from "components/Loader"
import StockDetailsLoader from "components/Stocks/DetailsPageItems/StockDetailsLoader"

// homepage route
const Homepage = Loadable({
  loader: () =>
    import("components/Home/Homepage" /* webpackChunkName: "Homepage" */),
  loading: Loader,
})

// authentication routes
const Login = Loadable({
  loader: () =>
    import("components/authentication/Login" /* webpackChunkName: "Login" */),
  loading: Loader,
})
const OauthCallback = Loadable({
  loader: () =>
    import(
      "components/authentication/OauthCallback" /* webpackChunkName: "OauthCallback" */
    ),
  loading: Loader,
})
const AuthLoader = Loadable({
  loader: () =>
    import(
      "components/authentication/AuthLoader" /* webpackChunkName: "AuthLoader" */
    ),
  loading: Loader,
})
const Logout = Loadable({
  loader: () =>
    import("components/authentication/Logout" /* webpackChunkName: "Logout" */),
  loading: Loader,
})
const MyDsc = Loadable({
  loader: () => import("components/MyDsc" /* webpackChunkName: "MyDsc" */),
  loading: Loader,
})

// infopage routes
const InfoPage = Loadable({
  loader: () =>
    import("components/InfoPage/InfoPage" /* webpackChunkName: "InfoPage" */),
  loading: Loader,
})
const EditInfoPage = Loadable({
  loader: () =>
    import(
      "components/InfoPage/EditInfoPage" /* webpackChunkName: "EditInfoPage" */
    ),
  loading: Loader,
})

// strain routes
const StrainCatalogWrapper = Loadable({
  loader: () =>
    import(
      "components/Stocks/Strains/Catalog/StrainCatalogWrapper" /* webpackChunkName: "StrainCatalogWrapper" */
    ),
  loading: StockDetailsLoader,
})
const StrainDetailsContainer = Loadable({
  loader: () =>
    import(
      "components/Stocks/Strains/StrainDetailsContainer" /* webpackChunkName: "StrainDetailsContainer" */
    ),
  loading: StockDetailsLoader,
})

// plasmid routes
const PlasmidCatalogContainer = Loadable({
  loader: () =>
    import(
      "components/Stocks/Plasmids/Catalog/PlasmidCatalogContainer" /* webpackChunkName: "PlasmidCatalogContainer" */
    ),
  loading: StockDetailsLoader,
})
const PlasmidDetailsContainer = Loadable({
  loader: () =>
    import(
      "components/Stocks/Plasmids/PlasmidDetailsContainer" /* webpackChunkName: "PlasmidDetailsContainer" */
    ),
  loading: StockDetailsLoader,
})

// order form routes
const OrderForm = Loadable({
  loader: () =>
    import(
      "components/OrderForm/OrderForm" /* webpackChunkName: "OrderForm" */
    ),
  loading: Loader,
})
const OrderConfirmation = Loadable({
  loader: () =>
    import(
      "components/OrderForm/OrderConfirmation" /* webpackChunkName: "OrderConfirmation" */
    ),
  loading: Loader,
})

// misc routes
const Contact = Loadable({
  loader: () => import("components/Contact" /* webpackChunkName: "Contact" */),
  loading: Loader,
})
const ErrorPage = Loadable({
  loader: () =>
    import("components/Errors/ErrorPage" /* webpackChunkName: "ErrorPage" */),
  loading: Loader,
})
const ShoppingCartPage = Loadable({
  loader: () =>
    import(
      "components/ShoppingCart/ShoppingCartPage" /* webpackChunkName: "ShoppingCartPage" */
    ),
  loading: Loader,
})

// custom 404 route
const PageNotReady = Loadable({
  loader: () =>
    import(
      "components/Errors/PageNotReady" /* webpackChunkName: "PageNotReady" */
    ),
  loading: Loader,
})

const RenderRoutes = () => (
  <Switch>
    <Route exact path="/" component={Homepage} />
    {/* authentication routes */}
    <LoginRoute exact path="/login" component={Login} />
    <Route exact path="/:provider/callback" component={OauthCallback} />
    <Route exact path="/load/auth" component={AuthLoader} />
    <PrivateRoute exact path="/logout" component={Logout} />
    <PrivateRoute exact path="/mydsc" component={MyDsc} />
    {/* infopage routes */}
    <Route exact path="/information/:name" component={InfoPage} />
    <PrivateRoute
      exact
      path="/information/:name/edit"
      component={EditInfoPage}
    />
    {/* order form routes */}
    <Route exact path="/order/checkout" component={OrderForm} />
    <Route exact path="/order/submitted" component={OrderConfirmation} />
    {/* strain routes */}
    <Route exact path="/strains" component={StrainCatalogWrapper} />
    <Route exact path="/strains/:id" component={StrainDetailsContainer} />
    {/* plasmid routes */}
    <Route exact path="/plasmids" component={PlasmidCatalogContainer} />
    <Route exact path="/plasmids/:id" component={PlasmidDetailsContainer} />
    {/* misc routes */}
    <Route exact path="/contact" component={Contact} />
    <Route exact path="/error" component={ErrorPage} />
    <Route exact path="/cart" component={ShoppingCartPage} />
    <Route exact path="*" component={PageNotReady} />
  </Switch>
)

export default RenderRoutes
