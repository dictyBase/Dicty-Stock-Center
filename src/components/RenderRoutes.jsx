import React, { Component } from 'react'
import { Route, Switch } from 'react-router-dom'
import routes from '../routes'

export default class RenderRoutes extends Component {
    displayName = 'list of routes';
    render() {
        return (
            <Switch>
                { routes.map((route, i) => (
                    <Route
                        exact
                        key={ i }
                        path={ route.path }
                        render={ ({ match }) =>
                        <route.component {...this.props}
                        match={ match } /> }
                     />
                )) }
            </Switch>
        )
    }
}

