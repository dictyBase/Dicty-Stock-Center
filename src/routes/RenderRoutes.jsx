import React from "react"
import { Route, Switch } from "react-router-dom"
import Loadable from "react-loadable"
import PrivateRoute from "./PrivateRoute"
import LoginRoute from "./LoginRoute"
import Loader from "components/Loader"
import StockDetailsLoader from "components/Stocks/StockDetailsLoader"
// import InfoPage from "components/InfoPage/InfoPage"
// import EditInfoPage from "components/InfoPage/EditInfoPage"

// homepage route
const Homepage = Loadable({
  loader: () =>
    import("components/Home/Homepage" /* webpackChunkName: "Homepage" */),
  loading: Loader,
})

// authentication routes
const Login = Loadable({
  loader: () =>
    import("components/authentication/Login" /* webpackChunkName: "Login" */),
  loading: Loader,
})
const OauthCallback = Loadable({
  loader: () =>
    import("components/authentication/OauthCallback" /* webpackChunkName: "OauthCallback" */),
  loading: Loader,
})
const AuthLoader = Loadable({
  loader: () =>
    import("components/authentication/AuthLoader" /* webpackChunkName: "AuthLoader" */),
  loading: Loader,
})
const Logout = Loadable({
  loader: () =>
    import("components/authentication/Logout" /* webpackChunkName: "Logout" */),
  loading: Loader,
})
const MyDsc = Loadable({
  loader: () => import("components/MyDsc" /* webpackChunkName: "MyDsc" */),
  loading: Loader,
})

// infopage routes
const InfoPage = Loadable({
  loader: () =>
    import("components/InfoPage/InfoPage" /* webpackChunkName: "InfoPage" */),
  loading: Loader,
})
const EditInfoPage = Loadable({
  loader: () =>
    import("components/InfoPage/EditInfoPage" /* webpackChunkName: "EditInfoPage" */),
  loading: Loader,
})

// strain routes
const StrainCatalogContainer = Loadable({
  loader: () =>
    import("components/Stocks/Strains/StrainCatalogContainer" /* webpackChunkName: "StrainCatalogContainer" */),
  loading: StockDetailsLoader,
})
const StrainDetailsContainer = Loadable({
  loader: () =>
    import("components/Stocks/Strains/StrainDetailsContainer" /* webpackChunkName: "StrainDetailsContainer" */),
  loading: StockDetailsLoader,
})

// plasmid routes
const PlasmidCatalogContainer = Loadable({
  loader: () =>
    import("components/Stocks/Plasmids/PlasmidCatalogContainer" /* webpackChunkName: "PlasmidCatalogContainer" */),
  loading: Loader,
})
const PlasmidDetailsContainer = Loadable({
  loader: () =>
    import("components/Stocks/Plasmids/PlasmidDetailsContainer" /* webpackChunkName: "PlasmidDetailsContainer" */),
  loading: Loader,
})

// order form routes
const Shipping = Loadable({
  loader: () =>
    import("components/form/Shipping" /* webpackChunkName: "Shipping" */),
  loading: Loader,
})
const EditShipping = Loadable({
  loader: () =>
    import("components/form/EditShipping" /* webpackChunkName: "EditShipping" */),
  loading: Loader,
})
const Payment = Loadable({
  loader: () =>
    import("components/form/Payment" /* webpackChunkName: "Payment" */),
  loading: Loader,
})
const EditPayment = Loadable({
  loader: () =>
    import("components/form/EditPayment" /* webpackChunkName: "EditPayment" */),
  loading: Loader,
})
const Submit = Loadable({
  loader: () =>
    import("components/form/Submit" /* webpackChunkName: "Submit" */),
  loading: Loader,
})
const SubmitLoader = Loadable({
  loader: () =>
    import("components/form/SubmitLoader" /* webpackChunkName: "SubmitLoader" */),
  loading: Loader,
})
const OrderConfirmation = Loadable({
  loader: () =>
    import("components/OrderConfirmation" /* webpackChunkName: "OrderConfirmation" */),
  loading: Loader,
})

// misc routes
const Contact = Loadable({
  loader: () => import("components/Contact" /* webpackChunkName: "Contact" */),
  loading: Loader,
})
const ErrorPage = Loadable({
  loader: () =>
    import("components/ErrorPage" /* webpackChunkName: "ErrorPage" */),
  loading: Loader,
})
const ShoppingCart = Loadable({
  loader: () =>
    import("components/ShoppingCart" /* webpackChunkName: "ShoppingCart" */),
  loading: Loader,
})

// custom 404 route
const PageNotReady = Loadable({
  loader: () =>
    import("components/PageNotReady" /* webpackChunkName: "PageNotReady" */),
  loading: Loader,
})

const RenderRoutes = () => (
  <Switch>
    <Route exact path="/" component={Homepage} />
    {/* authentication routes */}
    <LoginRoute exact path="/login" component={Login} />
    <Route exact path="/:provider/callback" component={OauthCallback} />
    <Route exact path="/load/auth" component={AuthLoader} />
    <PrivateRoute exact path="/logout" component={Logout} />
    <PrivateRoute exact path="/mydsc" component={MyDsc} />
    {/* infopage routes */}
    <Route exact path="/information/:name" component={InfoPage} />
    <PrivateRoute
      exact
      path="/information/:name/edit"
      component={EditInfoPage}
    />
    {/* order form routes */}
    <Route exact path="/order/shipping" component={Shipping} />
    <Route exact path="/order/shipping/edit" component={EditShipping} />
    <Route exact path="/order/payment" component={Payment} />
    <Route exact path="/order/payment/edit" component={EditPayment} />
    <Route exact path="/order/submit" component={Submit} />
    <Route exact path="/order/submitting" component={SubmitLoader} />
    <Route exact path="/order/submitted" component={OrderConfirmation} />
    {/* strain routes */}
    <Route exact path="/strains" component={StrainCatalogContainer} />
    <Route exact path="/strains/:id" component={StrainDetailsContainer} />
    {/* plasmid routes */}
    <Route exact path="/plasmids" component={PlasmidCatalogContainer} />
    <Route exact path="/plasmids/:id" component={PlasmidDetailsContainer} />
    {/* misc routes */}
    <Route exact path="/contact" component={Contact} />
    <Route exact path="/error" component={ErrorPage} />
    <Route exact path="/cart" component={ShoppingCart} />
    <Route exact path="*" component={PageNotReady} />
  </Switch>
)

export default RenderRoutes
