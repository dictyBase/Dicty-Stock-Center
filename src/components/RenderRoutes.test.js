import React from 'react'
import { BrowserRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import RenderRoutes from './RenderRoutes'

test('matching a snapshot of RenderRoutes', () => {
    const component = renderer.create(
    <BrowserRouter>
        <RenderRoutes />
    </BrowserRouter>
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})
