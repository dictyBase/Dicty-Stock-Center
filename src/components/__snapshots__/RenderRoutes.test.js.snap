// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching a snapshot of RenderRoutes 1`] = `
<div
  className="sc-eerKOB gRmrxR"
>
  <div
    className="sc-gFaPwZ hPYVgC"
  >
    <h1>
      <span
        aria-hidden={true}
        className="fa fa-wrench"
      />
       Under Construction
    </h1>
    <p>
      This page is not ready yet.
    </p>
    <button
      className="sc-dEoRIm fXhViD"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Stock Center Home
      </a>
    </button>
  </div>
</div>
`;
